src/middlewared/middlewared/plugins/idmap.py

```diff
@@ -322,7 +322,7 @@ async def get_next_idmap_range(self):
     @private
     async def snapshot_samba4_dataset(self):
         sysdataset = (await self.middleware.call('systemdataset.config'))['basename']
-        ts = str(datetime.datetime.now(datetime.timezone.utc).timestamp())[:10]
```

Comment:
What's wrong with using the original method?

Reply:
I believe the rationale behind @yocalebo 's idea to write a time util function was to consolidate all of our `datetime.now` calls into one place since they're used so frequently in middleware. This makes it easier in the future to control how we get the current time.

Reply:
After realizing the nuances of working with `datetime` objects, I'm going to heed your warning and not touch the existing `datetime.now` calls...